plugins/woocommerce/src/Internal/Features/FeaturesController.php

```diff
@@ -166,11 +166,27 @@ private function get_feature_definitions() {
 			$legacy_features = array(
 				'analytics'            => array(
 					'name'               => __( 'Analytics', 'woocommerce' ),
-					'description'        => __( 'Enable WooCommerce Analytics', 'woocommerce' ),
+					'description'        => sprintf(
+						// translators: 1: line break tag.
+						__(
+							'Enable WooCommerce Analytics.%1$s
+							The legacy Reports code will soon be removed.
+							If you enable the currently maintained Analytics now, you will not be able to turn it back off.',
+							'woocommerce'
+						),
+						'<br/>'
+					),
 					'option_key'         => Analytics::TOGGLE_OPTION_NAME,
 					'is_experimental'    => false,
 					'enabled_by_default' => true,
-					'disable_ui'         => false,
+					// Do not show the setting at all, if that's completely fresh install.
+					'disable_ui'         => get_option( Analytics::TOGGLE_OPTION_NAME, null ) === null,
+					'setting'            => array(
+						// Disable turning the feature off.
+						'disabled' => function () {
+							return get_option( Analytics::TOGGLE_OPTION_NAME ) === 'yes';
```

Comment:
On one hand, I think that, as we don't have a detailed plan for the evolution of the current analytics, it's better not to rush things. On the other, I doubt we'd be aiming to support analytics-disabled environments.

Reply:
Since in addition to the data source for the Dashboard widget, this option doesn't make Analytics and legacy reports mutually exclusive, I think it's feasible.